### PR TITLE
Extended the InventoryBaseDSC class to resolve the path to a specific Inventory Item

### PR DESCRIPTION
**Pull request by SimeonGerginov**
Signed-off-by: SimeonGerginov <simeongerginov1@gmail.com>

### CODE REVIEW

**Comment by SimeonGerginov** [comment]
Tested with the following values:

**DatacenterPath** = **Datacenter**
**InventoryItemPath** = **host/MyNewFolder1/MyNewFolder2/MyNewFolder3/MyNewFolder4/MyTestCluster**
**InventoryItemName** = **Resources**

File: Source/VMware.vSphereDSC/Classes/InventoryBaseDSC.ps1 [hunk 1]
```diff
@@ -18,83 +18,188 @@ class InventoryBaseDSC : BaseDSC {
     <#
     .DESCRIPTION
 
+    Name of the Inventory Item from the Datacenter we will use.
```
Comment (dmilov) [inline Source/VMware.vSphereDSC/Classes/InventoryBaseDSC.ps1]: Name of the resource under the Datacenter of 'Datacenter' key property

File: Source/VMware.vSphereDSC/Classes/InventoryBaseDSC.ps1 [hunk 2]
```diff
@@ -18,83 +18,188 @@ class InventoryBaseDSC : BaseDSC {
...
+    #>
+    [DscProperty(Key)]
+    [string] $Name
+
+    <#
+    .DESCRIPTION
+
+    The full path to the Inventory Item in the Datacenter we will use from the Inventory.
```
Comment (dmilov) [inline Source/VMware.vSphereDSC/Classes/InventoryBaseDSC.ps1]: Inventory folder path location of the resource with name specified on Name key property in the Datacenter specified in the 'Datacenter' key property

File: Source/VMware.vSphereDSC/Classes/InventoryBaseDSC.ps1 [hunk 3]
```diff
@@ -18,83 +18,188 @@ class InventoryBaseDSC : BaseDSC {
...
+    The path consists of 0 or more folders.
```
Comment (dmilov) [inline Source/VMware.vSphereDSC/Classes/InventoryBaseDSC.ps1]: Empty path means the resource is in the root inventory folder.

File: Source/VMware.vSphereDSC/Classes/InventoryBaseDSC.ps1 [hunk 4]
```diff
@@ -18,83 +18,188 @@ class InventoryBaseDSC : BaseDSC {
...
+    The Root folders of the Datacenter are not part of the path.
+    The parts of the path are separated with "/".
```
Comment (dmilov) [inline Source/VMware.vSphereDSC/Classes/InventoryBaseDSC.ps1]: Folder names in path are separated by "/" "<parent folder name >/<child folder name>"

File: Source/VMware.vSphereDSC/Classes/InventoryBaseDSC.ps1 [hunk 5]
```diff
@@ -18,83 +18,188 @@ class InventoryBaseDSC : BaseDSC {
...
+    Example path: "<Folder Name>/<Folder Name>"
```
Comment (dmilov) [inline Source/VMware.vSphereDSC/Classes/InventoryBaseDSC.ps1]: Example path for a VM resource: "Discovered Virtual Machines/My Ubuntu VMs"

File: Source/VMware.vSphereDSC/Classes/InventoryBaseDSC.ps1 [hunk 6]
```diff
@@ -18,83 +18,188 @@ class InventoryBaseDSC : BaseDSC {
...
+    [DscProperty(Key)]
+    [string] $InventoryPath
+
+    <#
+    .DESCRIPTION
+
     The full path to the Datacenter we will use from the Inventory.
     The path consists of 0 or more folders and the Datacenter name.
-	Root folder of the Inventory is not part of the path.
+    Root folder of the Inventory is not part of the path.
```
Comment (dmilov) [inline Source/VMware.vSphereDSC/Classes/InventoryBaseDSC.ps1]: Root 'datacenters' folder is not part of the path. Path can't be empty. Last item in the path is the datacenter name. If only the datacenter name is specified datacenter will be searched under the root 'datacenters' folder

File: Source/VMware.vSphereDSC/Classes/InventoryBaseDSC.ps1 [hunk 7]
```diff
@@ -18,83 +18,188 @@ class InventoryBaseDSC : BaseDSC {
...
     The parts of the path are separated with "/" where the last part of the path is the Datacenter name.
     Example path: "<Folder Name>/<Folder Name>/<Datacenter Name>".
```
Comment (dmilov) [inline Source/VMware.vSphereDSC/Classes/InventoryBaseDSC.ps1]: Example path: "MyDatacentersFolder/MyDatacenter"

File: Source/VMware.vSphereDSC/Classes/InventoryBaseDSC.ps1 [hunk 8]
```diff
@@ -18,83 +18,188 @@ class InventoryBaseDSC : BaseDSC {
...
     #>
     [DscProperty(Key)]
-    [string] $DatacenterPath
+    [string] $Datacenter
 
     <#
     .DESCRIPTION
 
-    Value indicating if the Inventory object should be Present or Absent.
+    Value indicating if the Inventory Item should be Present or Absent.
```
Comment (dmilov) [inline Source/VMware.vSphereDSC/Classes/InventoryBaseDSC.ps1]: if the Resource

File: Source/VMware.vSphereDSC/Classes/InventoryBaseDSC.ps1 [hunk 9]
```diff
@@ -18,83 +18,188 @@ class InventoryBaseDSC : BaseDSC {
...
     #>
     [DscProperty(Mandatory)]
     [Ensure] $Ensure
 
     <#
     .DESCRIPTION
 
+    The type of Datacenter Folder in which the Inventory Item is located.
```
Comment (dmilov) [inline Source/VMware.vSphereDSC/Classes/InventoryBaseDSC.ps1]: in which the resource is located. Possible values are VM, Network, Datastore, Host

**Pull request merged**